databuilder/query_engines/in_memory.py

```diff
@@ -230,6 +230,12 @@ def visit_Subtract(self, node):
     def visit_Multiply(self, node):
         return self.visit_binary_op_with_null(node, operator.mul)
 
+    def visit_TrueDivide(self, node):
+        return self.visit_binary_op_with_null(node, operator.div)
+
+    def visit_FloorDivide(self, node):
+        return self.visit_binary_op_with_null(node, operator.floordiv)
```

Comment:
I think we'll end up having to define our own function here, rather than reusing the one from `operator`, because Python's floordiv behaviour is slightly different from the SQL behaviour we want to emulate. But we should just define the desired behaviour in the spec tests and let them drive the necessary changes out.

Reply:
> But we should just define the desired behaviour in the spec tests and let them drive the necessary changes out.

Thanks, don't understand where we would need to define our own `floordiv` function and how this relates to the spec tests? What do you mean by "_let them drive the necessary changes out_"?

Reply:
Sorry, that was spectacularly opaque on my part! I was referring to this part of the ticket:

> Another subtlety is that we want floor division for negative numbers to round towards zero, rather than downwards i.e. `-1 // 2 = 0`. This is in contrast to Python where the result would be `-1`. I actually prefer the Python behaviour here; but matching the default SQL behaviour makes the implementation much simpler, and I doubt many people care about the difference.

So what I meant was that we'll need to add a case to the spec tests to cover this behaviour e.g.

    -4 // 3 == -1

As soon as you add that test you'll discover that the in-memory engine fails, because Python rounds the result down to -2. And so you'll have to write a custom Python division function with different rounding behaviour to get the tests to pass. (This is what I meant by letting the tests drive the necessary changes out.)